galsim/catalog.py

```diff
@@ -312,3 +317,334 @@ def itervalues(self):
     def iteritems(self):
         return self.dict.iteritems()
 
+def makeCOSMOSCatalog(file_name, use_real=True, image_dir=None, dir=None, noise_dir=None,
+                      preload=False, deep_sample=False, exclude_fail=True):
+    """
+    This routine makes a catalog of galaxies based on the COSMOS sample with F814W<23.5.
+
+    Depending on the keyword arguments, particularly `use_real`, the catalog will either have
+    information about real galaxies, or parametric ones.  To use this with either type of galaxies,
+    you need to get the COSMOS datasets in the format that GalSim recognizes; see
+
+        https://github.com/GalSim-developers/GalSim/wiki/RealGalaxy-Data
```

Comment:
Update: I wrote the script for doing the download into the GalSim share directory.  

I haven't yet implemented using it if `file_name=None`.  I'll work on getting that working for the RealGalaxyCatalog class.  Then assuming you like the way that works, you can update this function's docs accordingly.